lib/omnibus/fetchers/net_fetcher.rb

```diff
@@ -130,6 +130,25 @@ def checksum
       source[digest_type]
     end
 
+    #
+    # The md5 equivalent checksum of the fetched assets.
+    #
+    # If the user provides an md5 in the software definition, then that is
+    # used. Otherwise, the downloaded file is verified using the user
+    # specified digest and a fresh md5 checksum is calculated.
+    #
+    # Remember to call this function with s3 caching disabled if you are
+    # attempting to generate this md5 to look up this object in s3.
+    #
+    # @return [String]
+    #
+    def md5
```

Comment:
This seems a little odd to me. A function like `md5` is very much a getter in brain. Having it go off and maybe download a file feels weird.

Related question:
What happens if your software definition only specifies a sha and then we try fetch that package from the cache? Are we going to download form the canonical source to get the hash and then from the s3 cache?